Clippy subtree update

flip1995 opened the pull request:
r? Manishearth

`Cargo.lock` update due to 28d62f083376f956f28b6b3910446a2f171959b8. It should not affect the Rust repo, as `clippy_dev` is not used in the Rust repo.

jdonszelmann committed:
EII ast changes

profetia committed:
fix: `branches-sharing-code` suggests wrongly on `const` and `static`

km274 committed:
Implement lint to check for usages of `from_raw_parts` where the same expression is used for the length and the capacity

profetia committed:
fix: `map_entry` suggests wrongly for insert with cfg-ed out code

ada4a committed:
feat(format_push_string): add a suggestion

ada4a committed:
clean-up

SLUCHABLUB committed:
Fix an incorrect error message regarding the size of `isize` and `usize`

ada4a committed:
dogfood the lint

This commit is put before the one that actually introduces the lint,
since, were they put in the opposite order, the first commit wouldn't
pass dogfood

profetia committed:
fix: `if_not_else` wrong unmangled macros

profetia committed:
fix: `set-contains-or-insert` FP when set is mutated before `insert`

profetia committed:
fix: `match_like_matches_macro` wrongly unmangled macros

alex-semenyuk committed:
Upgrade Node.js setup action and version

cjgillot committed:
Replace Rvalue::NullaryOp by a variant in mir::ConstValue.

meithecatte committed:
`search_is_some`: move to nursery

See clippy issue 16086 for context

cjgillot committed:
Introduce Operand::RuntimeChecks.

smoelius committed:
Allow multiline suggestions in `map-unwrap-or`

samueltardieu committed:
Count unsafe operations only towards the innermost unsafe block

Also, macro calls are counted as at most one unsafe operation.

BoxyUwU committed:
Fix tools

ada4a committed:
feat(unnecessary_fold): lint on `fold`s with `Add::add`/`Mul::mul`

ada4a committed:
clean-up

profetia committed:
fix: `empty_enum_variants_with_brackets` misses removing brackets in patterns

profetia committed:
Make `empty_enum_variants_with_brackets` to support empty variant with braces

profetia committed:
fix: `use_self` FP on type in const generics

samueltardieu committed:
Do not look for disallowed methods inside desugared code

GuillaumeGomez committed:
Update clippy code to new doc attribute API

Urgau committed:
Adapt Clippy to the overhaul filename handling

GuillaumeGomez committed:
Update clippy code

relaxcn committed:
Fix `if_then_some_else_none` FP when the `then` block contains `.await`

cuongleqq committed:
Add iterator reduction coverage to never_loop

Alexendoo committed:
Move `collapsible_else_if` to `pedantic`

Abd002 committed:
Fix clippy::ref_as_ptr for non-temporary references in let/const

RalfJung committed:
miri: add -Zbinary-dep-depinfo to dependency builds

flip1995 committed:
Merge commit '9e3e9649cbae7ed33ba62ca436f43081eaaedeb5' into clippy-subtree-update

bors committed:
Auto merge of #149645 - GuillaumeGomez:doc-attr-based, r=jdonszelmann,jonathanbrouwer

Port `doc` attributes to new attribute API

Part of https://github.com/rust-lang/rust/issues/131229.

This PR ports the `doc` attributes to the new attribute API. However, there are things that will need to be fixed in a follow-up:

* Some part of `cfg_old.rs` are likely unused now, so they should be removed.
* Not all error/lints are emitted at the same time anymore, making them kinda less useful considering that you need to run and fix rustc/rustdoc multiple times to get through all of them.
* For coherency with the other attribute errors, I didn't modify the default output too much, meaning that we have some new messages now. I'll likely come back to that to check if the previous ones were better in a case-by-case approach.
* `doc(test(attr(...)))` is handled in a horrifying manner currently. Until we can handle it correctly with the `Attribute` system, it'll remain that thing we're all very ashamed of. :smiling_imp:
* A type in rustdoc got its size increased, I'll check the impact on performance. But in any case, I plan to improve it in a follow-up so should be "ok".
* Because of error reporting, some fields of `Doc` are suboptimal, like `inline` which instead of being an `Option` is a `ThinVec` because we report the error later on. Part of the things I'm not super happy about but can be postponed to future me.
* In `src/librustdoc/clean/cfg.rs`, the `pub(crate) fn parse(cfg: &MetaItemInner) -> Result<Cfg, InvalidCfgError> {` function should be removed once `cfg_trace` has been ported to new `cfg` API.
* Size of type `DocFragment` went from 32 to 48. Would be nice to get it back to 32.
* ``malformed `doc` attribute input`` wasn't meant for so many candidates, should be improved.
* See how many of the checks in `check_attr` we can move to attribute parsing
* Port target checking to be in the attribute parser completely
* Fix target checking for `doc(alias)` on fields & patterns

And finally, once this PR is merged, I plan to finally stabilize `doc_cfg` feature. :)

cc `@jdonszelmann`
r? `@JonathanBrouwer`

cuongleqq committed:
Fix fmt and dogfood lints for never_loop iterator reduction

cuongleqq committed:
Clean up unused try_reduce entry in sym

matthiaskrgr committed:
Rollup merge of #149884 - flip1995:clippy-subtree-update, r=matthiaskrgr

Clippy subtree update

r? ``````@Manishearth``````

Cargo.lock update due to Clippy version bump.

Coca162 committed:
`transmuting_null`: Check const integer casts

changelog: [`transmuting_null`]: now checks const integers being casted
to pointers

Jarcho committed:
Fix `if_then_some_else_none` FP when encountering `await` codes (#16178)

changelog: Fix FP of [`if_then_some_else_none`] when the `then` block
contains `await` codes.

That is because `bool:then` doesn't work with await code.

Closes: rust-lang/rust-clippy#16176

Jarcho committed:
feat(unnecessary_fold): lint on `fold`s with `Add::add`/`Mul::mul` (#16124)

Resurrects https://github.com/rust-lang/rust-clippy/pull/13475

changelog: [`unnecessary_fold`]: lint on `fold`s with
`Add::add`/`Mul::mul`

bors committed:
Auto merge of #149709 - Urgau:overhaul-filenames, r=davidtwco

Overhaul filename handling for cross-compiler consistency

This PR overhauls the way we handle filenames in the compiler and `rmeta` in order to achieve achieve cross-compiler consistency (ie. having the same path no matter if the filename was created in the current compiler session or is coming from `rmeta`).

This is required as some parts of the compiler rely on consistent paths for the soundness of generated code (see rust-lang/rust#148328).

In order to achieved consistency multiple steps are being taken by this PR:
 - by making `RealFileName` immutable
 - by only having `SourceMap::to_real_filename` create `RealFileName`
   - currently `RealFileName` can be created from any `Path` and are remapped afterwards, which creates consistency issue
 - by also making `RealFileName` holds it's working directory, embeddable name and the remapped scopes
   - this removes the need for a `Session`, to know the current(!) scopes and cwd, which is invalid as they may not be equal to the scopes used when creating the filename

In order for `SourceMap::to_real_filename` to know which scopes to apply `FilePathMapping` now takes the current remapping scopes to apply, which makes `FileNameDisplayPreference` and company useless and are removed.

This PR is split-up in multiple commits (unfortunately not atomic), but should help review the changes.

Unblocks https://github.com/rust-lang/rust/pull/147611
Fixes https://github.com/rust-lang/rust/issues/148328

cuongleqq committed:
Refine never_loop iterator reduction check

Jarcho committed:
Fix clippy::ref_as_ptr for non-temporary references in let/const (#16214)

Fix `ref_as_ptr` false positives for temporary values and
let/static/const initializers.

Fixes rust-lang/rust-clippy#16220

changelog: [`ref_as_ptr`]: Only lint on non-temporary expressions)

profetia committed:
fix: `unchecked_time_subtraction` FN on `Ops::sub` method call

profetia committed:
fix: `unchecked_time_subtraction` wrongly unmangled macros

profetia committed:
fix: `manual_instant_elapsed` wrongly unmangled macros

Jarcho committed:
Fix `unchecked_time_subtraction` FN on `Ops::sub` method call (#16233)

Closes rust-lang/rust-clippy#16230
Closes rust-lang/rust-clippy#16234
Closes rust-lang/rust-clippy#16236

changelog: [`unchecked_time_subtraction`] fix FN on `Ops::sub` method
call

cuongleqq committed:
Refactor loops MethodCall handling into single match

dswij committed:
Fix `if_not_else` wrong unmangled macros (#15931)

Closes rust-lang/rust-clippy#15924

changelog: [`if_not_else`] fix wrongly unmangled macros

bors committed:
Auto merge of #146348 - jdonszelmann:eiiv3, r=lcnr,oli-obk

Externally implementable items

Supersedes https://github.com/rust-lang/rust/pull/140010
Tracking issue: https://github.com/rust-lang/rust/issues/125418

Getting started:

```rust
#![feature(eii)]

#[eii(eii1)]
pub fn decl1(x: u64)
// body optional (it's the default)
{
    println!("default {x}");
}

// in another crate, maybe
#[eii1]
pub fn decl2(x: u64) {
    println!("explicit {x}");
}

fn main() {
    decl1(4);
}
```

- tiny perf regression, underlying issue makes multiple things in the compiler slow, not just EII, planning to solve those separately.
- No codegen_gcc support, they don't have bindings for weak symbols yet but could
- No windows support yet for weak definitions

This PR merges the implementation of EII for just llvm + not windows, doesn't yet contain like a new panic handler implementation or alloc handler. With this implementation, it would support implementing the panic handler in terms of EII already since it requires no default implementation so no weak symbols

The PR has been open in various forms for about a year now, but I feel that having some implementation merged to build upon

cuongleqq committed:
loops: reuse iterator checks for unused_enumerate_index

Jarcho committed:
Add iterator reduction coverage to never_loop (#16222)

Fixes rust-lang/rust-clippy#16061.

Extend `never_loop` to also lint iterator reduction methods (e.g.
`for_each`, `try_for_each`, `fold`, `try_fold`, `reduce`, `all`, `any`)
when the closure’s body diverges (return type `!`). Add a UI test
`never_loop_iterator_reduction` to cover these cases.

Testing:
- TESTNAME=never_loop_iterator_reduction cargo uitest

changelog: [`never_loop`]: lint diverging iterator reduction closures
like `for_each` and `fold`

samueltardieu committed:
feat(format_push_string): give a (possibly incomplete) suggestion (#16093)

Resolves https://github.com/rust-lang/rust-clippy/issues/8824

Supersedes https://github.com/rust-lang/rust-clippy/pull/15828 -- I
wasn't able to get `TyCtxt::in_scope_traits` to work, so here's a more
basic implementation that just adds a note about the possible need to
import `std::fmt::Write`.

changelog: [`format_push_string`]: add a suggestion

samueltardieu committed:
Fix `match_like_matches_macro` wrongly unmangled macros (#16018)

Closes rust-lang/rust-clippy#16015

changelog: [`match_like_matches_macro`] fix wrongly unmangled macros

samueltardieu committed:
Fix `map_entry` suggests wrongly for insert with cfg-ed out code (#15800)

Closes rust-lang/rust-clippy#15781

changelog: [`map_entry`] fix wrong suggestions for insert with cfg-ed
out code

samueltardieu committed:
Fix `set-contains-or-insert` FP when set is mutated before `insert` (#16009)

Closes rust-lang/rust-clippy#15990

changelog: [`set-contains-or-insert`] fix FP when set is mutated before
`insert`

Jarcho committed:
Count unsafe operations and macro calls once towards the innermost unsafe block (#16117)

changelog: [`multiple_unsafe_ops_per_block`]: count unsafe operations
only towards the innermost unsafe block

Fixes rust-lang/rust-clippy#16116

r? Jarcho

blyxyas committed:
Fix `branches-sharing-code` suggests wrongly on `const` and `static` (#15522)

Closes rust-lang/rust-clippy#15347

changelog: [`branches-sharing-code`] fix wrong suggestions on `const`
and `static`

samueltardieu committed:
Parent of struct constructor is the struct itself

Also, the `Fn`/`AssocFn` cases were not tested anywhere, and according
to the comments inside `TyCtxt::res_generic_def_id()` could have made
the `TyCtxt::generics_of()` call also trigger an ICE.

samueltardieu committed:
Move `needless_type_cast` lint to nursery

The lint is new and already has several bugs filed against it, including
an ICE.

Jarcho committed:
Parent of struct constructor is the struct itself (#16245)

The `Fn`/`AssocFn` cases were not tested anywhere, and according to the
comments inside `TyCtxt::res_generic_def_id()` could have made the
`TyCtxt::generics_of()` call also trigger an ICE.

changelog: [`needless_type_cast`]: do not ICE on struct constructor

Fixes rust-lang/rust-clippy#16243

Centri3 committed:
Do not look for disallowed methods inside desugared code (#16186)

changelog: [`disallowed_methods`]: do not check compiler-generated code

Fixes rust-lang/rust-clippy#16185

llogiq committed:
Move `needless_type_cast` lint to nursery (#16246)

changelog: [`needless_type_cast`]: move into nursery until bugs are
ironed out

[Zulip
thread](https://rust-lang.zulipchat.com/#narrow/channel/257328-clippy/topic/Moving.20.60needless_type_cast.60.20to.20.60nursery.60)

Kivooeo committed:
remove fixme & update stderr files

JonathanBrouwer committed:
Rollup merge of #150032 - Kivooeo:annotate-snippets-stable, r=Muscraft

Use annotate-snippet as default emitter on stable

This is implementation of https://github.com/rust-lang/rust/issues/149932

Now, after MCP was accepted, we can use annotate-snippet as default emitter for errors, that means that we not longer need of previous emitter, so this PR removed previous emitter and makes annotate-snippet new default one both on stable and nightly

(this PR does not remove a code of previous emitter it just removes a `Default` option of `HumanReadableErrorType` enum, and keeping only `HumanReadableErrorType::AnnotateSnippet` as it now uses by default)

Jarcho committed:
Move `collapsible_else_if` to `pedantic` (#16211)

Looks like a good number of people disagree with this lint so it
shouldn't be on by default

Fixes https://github.com/rust-lang/rust-clippy/issues/16209

changelog: [`collapsible_else_if`]: move to `pedantic`

samueltardieu committed:
Fix `empty_enum_variants_with_brackets` misses removing brackets in patterns (#16160)

Closes rust-lang/rust-clippy#16157

changelog: [`empty_enum_variants_with_brackets`] fix missing to remove
brackets in patterns

ada4a committed:
feat(repeated_is_diagnostic_item): new internal lint

samueltardieu committed:
New internal lint: `repeated_is_diagnostic_item` (#15937)

As described in
https://github.com/rust-lang/rust-clippy/pull/15528#issuecomment-3214018887

changelog: none

borngraced committed:
skip `needless_type_cast` for single-use non-coercible casts

llogiq committed:
Fix `needless_type_cast` suggesting invalid code for non-literal initializers (#16248)

When the initializer is a function call or suffixed literal (e.g.,
`size_of::<T>()` or `10u8`), suggest casting the initializer too, since
changing only the type annotation would fail to compile. Fixes
rust-lang/rust-clippy#16240

```
changelog: [`needless_type_cast`]: Fix suggesting invalid code for non-literal initializers
```

llogiq committed:
New lint - `same_length_and_capacity` (#15656)

Fixes rust-lang/rust-clippy#5955

I understand that there's a feature freeze right now. I had started
working on this before I was aware of the feature freeze. I don't mind
waiting as long as I need to for feedback.

changelog: [`same_length_and_capacity`]: adds a new lint that checks for
usages of `from_raw_parts` where the same expression is passed for the
length and the capacity

<!-- TRIAGEBOT_START -->

<!-- TRIAGEBOT_SUMMARY_START -->

### Summary Notes

-
[Feature-freeze](https://github.com/rust-lang/rust-clippy/pull/15656#issuecomment-3276620004)
by [github-actions[bot]](https://github.com/github-actions[bot])

*Managed by `@rustbot`—see
[help](https://forge.rust-lang.org/triagebot/note.html) for details*

<!-- TRIAGEBOT_SUMMARY_END -->
<!-- TRIAGEBOT_END -->

dswij committed:
`transmuting_null`: Check const integer casts (#16227)

Additionally removed the comment about MIR const propagation as it is
now outdated

changelog: [`transmuting_null`]: now checks const integers being casted
to pointers

llogiq committed:
Allow multiline suggestions in `map-unwrap-or` (#16114)

I recently got a `map-unwrap-or` warning with no suggestion, and was
very confused because I couldn't tell what the lint was flagging.

It turned out the "else" part had multiple lines, so the lint decided
not to include a suggestion.

Please look at this PR's stderr file, and decide whether you like it.
Then look at [its
predecessor](https://github.com/rust-lang/rust-clippy/blob/24e16f992c2dda1904fe8d4a94b391513832ce60/tests/ui/map_unwrap_or.stderr),
and decide whether you like it.

IMHO, the one with the suggestions is much more informative, even if not
as pretty.

changelog: Allow multiline suggestions in `map-unwrap-or`

relaxcn committed:
Fix `println_empty_string` suggestion caused error when there is a `,` after arg.

Make `writeln_empty_string` don't provide an auto-fix suggestion when
there is a comment in the macro call span.

samueltardieu committed:
Use actions/checkout@v6 consistently

Two places were not updated from v5 to v6.

samueltardieu committed:
Reenable some tests on "apple" targets

Those tests appear to be passing on "apple" targets as well.

llogiq committed:
Reenable some tests on "apple" targets (#16264)

Those tests appear to be passing on "apple" targets as well.

changelog: none

samueltardieu committed:
Set myself on vacation

samueltardieu committed:
Set myself on vacation (#16266)

Expecting a slow rythm until about Jan. 2.

changelog: none
r? ghost

samueltardieu committed:
Fix an incorrect error message regarding the size of `usize` and `isize` in `cast_precision_loss`. (#14966)

When trying to cast a `usize` or `isize` using `as` to a type of
potentially smaller size the `cast_precision_loss` would claim that the
size of `usize` is "32 or 64 bits". It is now corrected to "16, 32, or
64 bits".

changelog: [`cast_precision_loss`]: fix the error messages claim
regarding the size of `usize` and `isize`.

samueltardieu committed:
Fix `println_empty_string` suggestion caused error (#16201)

Closes: rust-lang/rust-clippy#16167
changelog: [`println_empty_string`]: fix suggestion caused error when
there is a comma after arg.

Closes: rust-lang/rust-clippy#16251
changelog: [`writeln_empty_string`]: fix suggestion caused error when
there is a comma in the comment before arg.

samueltardieu committed:
Do not attempt to use `nth` with non-usize argument

samueltardieu committed:
Preserve explicit lifetime information when removing `mut`

Rather than trying to reconstruct the reference type without the `mut`,
remove the `mut` itself so that user-inserted lifetime information is
preserved.

The message has been changed from "consider changing to [NON-MUT REF]"
to "remove this `mut`"

llogiq committed:
Do not attempt to use `nth` with non-usize argument (#16272)

Fixes rust-lang/rust-clippy#16270

----

changelog: [`needless_collect`]: do not attempt to suggest `nth` when
argument is not a `usize`

profetia committed:
fix: `result_large_err` FN on closures

profetia committed:
fix: `if_then_some_else_none` suggests wrongly when then ends with comment

dswij committed:
Fix `if_then_some_else_none` suggests wrongly when then ends with comment (#16278)

Closes rust-lang/rust-clippy#16269

changelog: [`if_then_some_else_none`] fix wrong suggestions when then
ends with comment

matthiaskrgr committed:
Rollup merge of #149882 - RalfJung:miri-dep-build, r=Mark-Simulacrum

miri: add -Zbinary-dep-depinfo to dependency builds

Hopefully fixes https://github.com/rust-lang/rust/issues/149711

Cc ````@bjorn3````

samueltardieu committed:
Use "parameter" instead of "argument" in the lint message

profetia committed:
`cmp_null`: add missing parens and mention of `ptr::null_mut` in its docs

profetia committed:
fix: `cmp_null` wrongly unmangled macros

Jarcho committed:
Fix `cmp_null` missing parens in the example (#16282)

Closes rust-lang/rust-clippy#16281

changelog: [`cmp_null`] Don't suggest to inline macro contents instead
of calling them

bors committed:
Auto merge of #148766 - cjgillot:mir-const-runtime-checks, r=RalfJung,saethlin

Replace Rvalue::NullaryOp by a variant in mir::Operand.

Based on https://github.com/rust-lang/rust/pull/148151

This PR fully removes the MIR `Rvalue::NullaryOp`. After rust-lang/rust#148151, it was only useful for runtime checks like `ub_checks`, `contract_checks` and `overflow_checks`.

These are "runtime" checks, boolean constants that may only be `true` in codegen. It depends on a rustc flag passed to codegen, so we need to represent those flags cross-crate.

This PR replaces those runtime checks by special variants in MIR `ConstValue`. This allows code that expects constants to manipulate those as such, even if we may not always be able to evaluate them to actual scalars.

samueltardieu committed:
Fix `use_self` FP on type in const generics (#16172)

Closes rust-lang/rust-clippy#16164

changelog: [`use_self`] fix FP on type in const generics

rwardd committed:
fix: remove else-if collapse suggestion when all branches contain only an `if {..} else {..}` expression

If two `if` arms contain only a single `if {..} else {..}` statement,
don't suggest a collapse as this can lead to less readable code.

profetia committed:
fix: `obfuscated_if_else` wrongly unmangled macros

dswij committed:
Don't emit `collapsible_else_if` lint when all arms contain only `if {} else {}` expressions (#16286)

If two `if` arms contain only a single `if {..} else {..}` statement,
don't suggest a collapse as this can lead to less readable code.

Fixes rust-lang/rust-clippy#4971

changelog: [`collapsible_else_if`]: prevent emitting lint when the all
`if` arms contain only an `if {..} else {..}` expression.

Jarcho committed:
Fix `obfuscated_if_else` wrongly unmangled macros (#16289)

Closes rust-lang/rust-clippy#16288

changelog: [`obfuscated_if_else`] fix wrongly unmangled macros

Jarcho committed:
Fix `result_large_err` FN on closures (#16277)

Closes rust-lang/rust-clippy#16249

changelog: [`result_large_err`] fix FN on closures

Jarcho committed:
Preserve explicit lifetime information when removing `mut` (#16273)

Rather than trying to reconstruct the reference type without the `mut`,
remove the `mut` itself so that user-inserted lifetime information is
preserved.

The message has been changed from "consider changing to [NON-MUT REF]"
to "remove this `mut`"

changelog: [`needless_pass_by_ref_mut`]: preserve user-provided lifetime
information

Fixes rust-lang/rust-clippy#16267

relaxcn committed:
Fix `assertions_on_constants` false positive when there is non-constant value in the condition expr

llogiq committed:
Fix `assertions_on_constants` FP when there is non-constant value in the condition expr (#16297)

Closes: rust-lang/rust-clippy#16242

----

changelog: [`assertions_on_constants`]: Fix false positive when there is
non-constant value in the condition expr.

flip1995 committed:
Merge remote-tracking branch 'upstream/master' into rustup

flip1995 committed:
Use actions/checkout@v6 consistently (#16263)

Two places were not updated from v5 to v6.

changelog: none
r? flip1995

flip1995 committed:
Upgrade Node.js setup action and version (#16039)

Upgrade Node.js setup action and version at remark workflow

changelog: none

rust-log-analyzer commented:
The job **`x86_64-gnu-tools`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/20507452987/job/58924063904) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/58924063904) [(plain)](https://github.com/rust-lang/rust/commit/634d5e4853c5374a67137e2795371aedbcc43295/checks/58924063904/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
tests/ui/should_impl_trait/method_list_1.rs (revision `edition2021`) ... ok
tests/ui/size_of_in_element_count/functions.rs ... ok

FAILED TEST: tests/ui/track-diagnostics.rs
command: CLIPPY_CONF_DIR="tests" RUSTC_ICE="0" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-linux-gnu/release/clippy-driver" "--error-format=json" "--emit=metadata" "-Aunused" "-Ainternal_features" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Dwarnings" "-Ldependency=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/release/deps" "--sysroot=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui_test/0/tests/ui" "tests/ui/track-diagnostics.rs" "-Z" "track-diagnostics" "--extern" "futures=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-linux-gnu/debug/deps/libfutures-a9034e1031764ee4.rlib" "--extern" "futures=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-linux-gnu/debug/deps/libfutures-a9034e1031764ee4.rmeta" "--extern" "itertools=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-linux-gnu/debug/deps/libitertools-49215d6c78da5485.rlib" "--extern" "itertools=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-linux-gnu/debug/deps/libitertools-49215d6c78da5485.rmeta" "--extern" "libc=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-linux-gnu/debug/deps/liblibc-647c6467502a5c31.rlib" "--extern" "libc=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-linux-gnu/debug/deps/liblibc-647c6467502a5c31.rmeta" "--extern" "parking_lot=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-linux-gnu/debug/deps/libparking_lot-e492cfb623514f4f.rlib" "--extern" "parking_lot=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-linux-gnu/debug/deps/libparking_lot-e492cfb623514f4f.rmeta" "--extern" "quote=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-linux-gnu/debug/deps/libquote-5c45143607600b30.rlib" "--extern" "quote=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-linux-gnu/debug/deps/libquote-5c45143607600b30.rmeta" "--extern" "regex=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-linux-gnu/debug/deps/libregex-6ecd666cd642c024.rlib" "--extern" "regex=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-linux-gnu/debug/deps/libregex-6ecd666cd642c024.rmeta" "--extern" "serde=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-linux-gnu/debug/deps/libserde-b4b7e251597dd4a9.rlib" "--extern" "serde=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-linux-gnu/debug/deps/libserde-b4b7e251597dd4a9.rmeta" "--extern" "syn=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-linux-gnu/debug/deps/libsyn-ba360121f8a9307c.rlib" "--extern" "syn=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-linux-gnu/debug/deps/libsyn-ba360121f8a9307c.rmeta" "--extern" "tokio=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-linux-gnu/debug/deps/libtokio-6ddc89563990f085.rlib" "--extern" "tokio=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-linux-gnu/debug/deps/libtokio-6ddc89563990f085.rmeta" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/debug/deps" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/x86_64-unknown-linux-gnu/debug/deps" "--edition" "2024"

error: actual output differed from expected
Execute `./x test src/tools/clippy --bless` to update `tests/ui/track-diagnostics.stderr` to the actual output
--- tests/ui/track-diagnostics.stderr
+++ <stderr output>
 error[E0308]: mismatched types
   --> tests/ui/track-diagnostics.rs:LL:CC
... 2 lines skipped ...
    |              ^ expected `A`, found `B`
    |
-   = note: -Ztrack-diagnostics: created at /rustc-dev/<sha>/compiler/rustc_trait_selection/src/error_reporting/infer/mod.rs:LL:CC
+   = note: -Ztrack-diagnostics: created at compiler/rustc_trait_selection/src/error_reporting/infer/mod.rs:LL:CC
 
 error: aborting due to 1 previous error
 
 For more information about this error, try `rustc --explain E0308`.

```

</details>

flip1995 committed:
Bump nightly version -> 2025-12-25

flip1995 committed:
Don't take rustc-literal-escaper from sysroot

This resulted in the following error for me:

    error[E0464]: multiple candidates for `rlib` dependency `rustc_literal_escaper` found
      --> clippy_dev/src/lib.rs:25:1
       |
    25 | extern crate rustc_literal_escaper;
       | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
       |
       = note: candidate #1: /home/pkrones/.rustup/toolchains/nightly-2025-12-25-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/librustc_literal_escaper-4f30574e01c3dad1.rlib
       = note: candidate #2: /home/pkrones/.rustup/toolchains/nightly-2025-12-25-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/librustc_literal_escaper-c8a0e29ad1040cee.rmeta

Other tools in the Rust repo also take that as an explicit dependency, like rust-analyzer or lint-docs.

flip1995 committed:
Rustup (#16299)

r? @ghost

changelog: none

flip1995 committed:
Merge commit '99edcadfd5f6f6e8da34b1ba62774b53f5ca3863' into clippy-subtree-update

flip1995 committed:
Update Cargo.lock

Manishearth commented:
@bors r+ p=1 rollup=never
